### PR TITLE
Update RAV samples to work with `fetchWith` create option

### PR DESCRIPTION
**Pull request by CobyPear**
Update RAV sample apps to work with `--fetchWith rest | graphql` when using `jss-create`
RAV should now work with Experience Edge out of the box.
## Description / Motivation

The Next sample already supports this, now RAV does too! 🎉 

As a JSS developer, I want to specify REST or GraphQL when using `jss-create`, so that I have less manual steps to get up and running with GraphQL

## Testing Details

Tested `jss-create` with the `--fetchWith graphql` option for React, Angular, and Vue

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: samples/react/sitecore/config/JssReactWeb.config [hunk 1]
```diff
@@ -29,13 +31,7 @@
         but disables workflow and publishing. Before going to production, change the `database`
         below to `web` instead of `master`.
       -->
-      <site patch:before="site[@name='website']"
```
Comment (ambrauer) [inline samples/react/sitecore/config/JssReactWeb.config:32]: Same here (revert formatting)

File: samples/vue/src/AppRoot.vue [hunk 2]
```diff
@@ -29,7 +29,7 @@ export default {
         // to show a loading/switching screen when language is being changed.
         this.languageIsChanging = true;
 
-        dictionaryService.fetchDictionaryData(language).then((phrases) => {
+        dictionaryServiceFactory.fetchDictionaryData(language).then((phrases) => {
```
Comment (ambrauer) [inline samples/vue/src/AppRoot.vue:32]: Don't we need to create a dictionary service instance first?
```suggestion
        const dictionaryServiceInstance = dictionaryServiceFactory.create();
        dictionaryServiceInstance.fetchDictionaryData(language).then((phrases) => {
```

**Review by ambrauer — CHANGES_REQUESTED** [review]
Looking good! See comments.

File: samples/angular/sitecore/config/JssAngularWeb.config [hunk 3]
```diff
@@ -29,13 +31,7 @@
...
         below to `web` instead of `master`.
       -->
-      <site patch:before="site[@name='website']"
-            inherits="website"
-            name="JssAngularWeb"
-            hostName="JssAngularWeb"
-            rootPath="/sitecore/content/JssAngularWeb"
-            startItem="/home"
-            database="master" />
+      <site patch:before="site[@name='website']" inherits="website" name="JssAngularWeb" hostName="JssAngularWeb" rootPath="/sitecore/content/JssAngularWeb" startItem="/home" database="master" />
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply samples/angular/sitecore/config/JssAngularWeb.config:34]: Absolutely. I blame prettier :p

File: samples/angular/src/app/jss-graphql.module.ts [hunk 4]
```diff
@@ -59,6 +59,10 @@ export class GraphQLModule {
       link passes cookies along with requests (withCredentials: true).
     */
 
+
+    // must send sc_apikey header when using GraphQL dictionary and layout service
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply samples/angular/src/app/jss-graphql.module.ts]: Ah right, fixed.

File: samples/angular/sitecore/config/JssAngularWeb.config [hunk 5]
```diff
@@ -7,28 +7,27 @@
...
-        PROXY CONFIGURATION
+      <!-- PROXY CONFIGURATION
         When running JSS with the headless proxy (node-headless-ssr-proxy),
         the original IP address of clients is obscured by the proxy.
         This setting tells Sitecore to read the forwarded header the proxy sends,
-        thus making analytics track the correct original client IP address.
-      -->
-      <setting name="Analytics.ForwardedRequestHttpHeader" set:value="X-Forwarded-For" />
+        thus making analytics track the correct original client IP address. -->
+      <setting name="Analytics.ForwardedRequestHttpHeader"
```
Comment (ambrauer) [inline samples/angular/sitecore/config/JssAngularWeb.config:20]: Looks like Prettier is still mashing the config formatting a bit. This `<setting />` shouldn't be multi-line, and then for all of the block comments, I'd stick with how they were (kind of a Sitecore standard) with
```
<!--
  Heading
  Description ...
  ...
-->
```
vs
```
<!-- Heading
  Description ...
  ... -->
```

**Review by ambrauer — APPROVED** [review]
Please see comment re: config formatting, otherwise this looks great!

File: samples/angular/sitecore/config/JssAngularWeb.config (hunk 5 above)
Comment (CobyPear) [reply samples/angular/sitecore/config/JssAngularWeb.config]: Noted. XML gods have been appeased.

**Pull request merged**